src/sources/soundsourceffmpeg.cpp

```diff
@@ -364,39 +403,61 @@ bool SoundSourceFFmpeg::readFramesToCache(unsigned int count, SINT offset) {
     }
 }
 
-bool SoundSourceFFmpeg::getBytesFromCache(char *buffer, SINT offset,
+bool SoundSourceFFmpeg::getBytesFromCache(CSAMPLE* buffer, SINT offset,
         SINT size) {
     struct ffmpegCacheObject *l_SObj = NULL;
-    quint32 l_lPos = 0;
+    qint32 l_lPos = 0;
     quint32 l_lLeft = 0;
     quint32 l_lOffset = 0;
     quint32 l_lBytesToCopy = 0;
 
+    char *l_pBuffer = (char *)buffer;
+
     // Is offset bigger than start of cache
     if (offset >= m_lCacheStartFrame) {
+        int l_lTmpLen = 0;
         // If last pos is (which is shouldn't) use caches end
         if (m_lCacheLastPos == 0) {
             m_lCacheLastPos = m_SCache.size() - 1;
         }
 
-        // Seek to correct FrameIndex
-        for (l_lPos = m_lCacheLastPos; l_lPos > 0; l_lPos --) {
+        // Seek to correct FrameIndex (Minus 5 for faster seek)
+        //
+        // This could be done per steps but because there
+        // Jump points can be far away and codec frames are small
+        // just jump to point where is safe to start.
+        for (l_lPos = m_lCacheLastPos; l_lPos >= 0; l_lPos -= 5) {
             l_SObj = m_SCache[l_lPos];
-            if ((l_SObj->startFrame + l_SObj->length) < offset) {
+
+            // Because length is in byte we have to convert it to Frames
+            l_lTmpLen = l_SObj->length / AUDIOSOURCEFFMPEG_MIXXXFRAME_TO_BYTEOFFSET;
+
+            if ((l_SObj->startFrame + l_lTmpLen) < offset) {
                 break;
             }
         }
 
+        // Because we step 5 backward we can end up to below zero
+        // We can't go futher so hope for the best
+        if (l_lPos < 0) {
+            l_lPos = 0;
+        }
+
         // Use this Cache object as starting point
         l_SObj = m_SCache[l_lPos];
 
+        if (l_pBuffer == NULL) {
+            qDebug() << "SoundSourceFFmpeg::getBytesFromCache: Out buffer NULL";
+            return false;
+        }
+
         // Calculate in other words get bytes how much we must copy to
         // buffer (CSAMPLE = 4 and we have 2 channels which is 8 times)
-        l_lLeft = (size * sizeof(CSAMPLE)) * 2;
-        memset(buffer, 0x00, l_lLeft);
+        l_lLeft = size * AUDIOSOURCEFFMPEG_MIXXXFRAME_TO_BYTEOFFSET;
+        memset(l_pBuffer, 0x00, l_lLeft);
```

Comment:
If you mean memset. Yes it's very redundant. Just an habit to clean everything. Is you mean l_lLeft then you incorrect.